src/lib/BaseClient.ts

```diff
@@ -12,12 +13,15 @@ export type BaseClientConfig = {
   retries?: number
   /** The delay between retries if the request fails */
   retryDelay?: number
+  /** The status numbers that would not be retried */
+  nonRetryableStatuses?: number[]
 }
 
 export abstract class BaseClient {
   private readonly baseUrl: string
   private readonly retries: number
   private readonly retryDelay: number
+  private readonly nonRetryableStatuses: number[]
```

Comment:
should this logic be added to the `BaseAPI.ts` too?

Reply:
I think we shouldn't because it would be interesting to start deprecating the `BaseAPI` class.
The idea behind the `BaseClient` is to be a way to move away from Axios to the authenticated fetch from the `decentraland-crypto-fetch`. This fetch contains the logic to perform authenticated requests and uses the fetch interface for making requests, making it more compatible with other components that we're building.
We can move features that are in the `BaseAPI` class that we don't have here though.